### DISCUSSION TITLE
Validate file and set a flag on record if the file passes validation

### DISCUSSION THREAD

**Discussion by DigNZ** [pocketbase/pocketbase#2614] [Q&A] [answered]
I'm struggling with this a little.

My record has a file along with other fields. The file is a mp4 video and needs to meet certain criteria to be automatically accepted - but could be overriden by an admin to force it to be accepted.

I have OnModelAfterCreate and OnModelAfterUpdate hooks because before update I found that file wasn't available.

I then call validateVideoFile and if it returns with an error - the error is logged but the Record is saved anyway. This is actually ok, but if I capture the error and use that to set a boolean on the record the change isn't saved because its on AfterCreate/Update

How do I either access the file before Create/Update OR write a value to the record from AfterCreate/AfterUpdate hook of that same record's action?

**Comment by ganigeorgiev** [comment, marked as answer]
You can't access the file in a `OnModelBefore*` hook because the upload happens after all the app model before hook listener has been executed and before the actual model persistence (this is mainly to allow changing for example the id of the record, which is part of the full file key).

If you want to access the file in a before hook, you can use `OnRecordBefore*` hook and check the `e.UploadedFiles` map.

`OnModelAfter*` model hooks are fired after the db persistence and if you want to apply another model change you'll have to save it manually. Eventually you can also create a new inline `Dao` to avoid triggering infinity recursion in the `OnModelUpdate*` hook, something like:

```go
app.OnModelAfterCreate("example").Add(func(e *core.ModelEvent) error {
	record := e.Model.(*models.Record)

	fileKey := record.BaseFilesPath() + "/" + record.GetString("yourFileField")

	fsys, err := app.NewFilesystem()
	if err != nil {
		return err
	}
	defer fsys.Close()

	file, err := fsys.GetFile(fileKey)
	if err != nil {
		return err
	}
	defer file.Close()

	if err := validateVideoFile(file); err != nil {
		return err
	}

	record.Set("someFlag", true)

	// create new dao from the current one but without the hooks
	// (there will be a more clearly named helper in the next release) 
	dao := daos.NewMultiDB(app.Dao().ConcurrentDB(), app.Dao().NonconcurrentDB())

	return dao.SaveRecord(record);
})
```

_**update**: added file key "/" delimiter._

**Reply by DigNZ** [reply]
Thanks!
